src/node_osrm.cpp

```diff
@@ -1,1161 +1,173 @@
-                            "(-180 < lng < 180, -90 < lat < 90)");
-            return resulting_coordinates;
-        }
+    const auto fn = Nan::GetFunction(fnTp).ToLocalChecked();
 
-        temp_coordinates.emplace_back(osrm::util::FloatLongitude(std::move(lon)),
-                                      osrm::util::FloatLatitude(std::move(lat)));
-    }
+    constructor().Reset(fn);
 
-    resulting_coordinates = boost::make_optional(std::move(temp_coordinates));
-    return resulting_coordinates;
+    Nan::Set(target, whoami, fn);
 }
 
-// Parses all the non-service specific parameters
-template <typename ParamType>
-bool argumentsToParameter(const Nan::FunctionCallbackInfo<v8::Value> &args,
-                          ParamType &params,
-                          bool requires_multiple_coordinates)
+NAN_METHOD(Engine::New)
 {
-    Nan::HandleScope scope;
-
-    if (args.Length() < 2)
-    {
-        Nan::ThrowTypeError("Two arguments required");
-        return false;
-    }
-
-    if (!args[0]->IsObject())
-    {
-        Nan::ThrowTypeError("First arg must be an object");
-        return false;
-    }
-
-    v8::Local<v8::Object> obj = Nan::To<v8::Object>(args[0]).ToLocalChecked();
-
-    v8::Local<v8::Value> coordinates = obj->Get(Nan::New("coordinates").ToLocalChecked());
-    if (coordinates->IsUndefined())
-    {
-        Nan::ThrowError("Must provide a coordinates property");
-        return false;
-    }
-    else if (coordinates->IsArray())
-    {
-        auto coordinates_array = v8::Local<v8::Array>::Cast(coordinates);
-        if (coordinates_array->Length() < 2 && requires_multiple_coordinates)
-        {
-            Nan::ThrowError("At least two coordinates must be provided");
-            return false;
-        }
-        else if (!requires_multiple_coordinates && coordinates_array->Length() != 1)
-        {
-            Nan::ThrowError("Exactly one coordinate pair must be provided");
-            return false;
-        }
-        auto maybe_coordinates = parseCoordinateArray(coordinates_array);
-        if (maybe_coordinates)
-        {
-            std::copy(maybe_coordinates->begin(), maybe_coordinates->end(),
-                      std::back_inserter(params->coordinates));
-        }
-        else
-        {
-            return false;
-        }
-    }
-    else if (!coordinates->IsUndefined())
-    {
-        BOOST_ASSERT(!coordinates->IsArray());
-        Nan::ThrowError("Coordinates must be an array of (lon/lat) pairs");
-        return false;
-    }
-
-    if (obj->Has(Nan::New("bearings").ToLocalChecked()))
+    if (info.IsConstructCall())
     {
-        v8::Local<v8::Value> bearings = obj->Get(Nan::New("bearings").ToLocalChecked());
-
-        if (!bearings->IsArray())
+        try
         {
-            Nan::ThrowError("Bearings must be an array of arrays of numbers");
-            return false;
-        }
+            auto config = argumentsToEngineConfig(info);
+            if (!config)
+                return;
```

Comment:
Why do we simply return here? (Was already in the old code like that)
Shouldn't this throw an error? This is done multiple times if I recall correctly.